bindings/python/client.py

```diff
@@ -153,15 +153,16 @@ def print_download_queue(console, download_queue):
 
     write_line(console, out)
 
+
 def add_torrent(ses, filename, options):
     atp = lt.add_torrent_params()
     if filename.startswith('magnet:'):
         atp = lt.parse_magnet_uri(filename)
     else:
         atp.ti = lt.torrent_info(filename)
         try:
-            at.resume_data = open(os.path.join(options.save_path, info.name() + '.fastresume'), 'rb').read()
-        except:
+            atp.resume_data = open(os.path.join(options.save_path, atp.info.name() + '.fastresume'), 'rb').read()
+        except BaseException:
```

Comment:
is this a style convention or is it a correctness issue?

Reply:
Some behaviour changed with bare except clauses in newer versions of python, basically a bare except will now act as 'except Exception:' which has a different behaviour than 'except BaseException:' (there are a few exceptions that are a direct subclass of BaseException). In the past there was no BaseException and everything was a subclass of Exception.

In order to not change the behaviour of the code an except BaseException was added here by autopep8. See https://www.python.org/dev/peps/pep-0348/ for some more details on the change and the hierarchy of Exceptions.

In this case, it's probably best to just catch Exception which was probably the original intent. In general, it is discouraged to ever use a blank except as a convention of style.

https://docs.python.org/3.1/howto/doanddont.html#except